internal/command/postgres/backup.go

```diff
@@ -111,17 +102,30 @@ func runBackupRestore(ctx context.Context) error {
 		return fmt.Errorf("No active machines")
 	}
 
-	// Ensure the the app has the required flex version.
-	if err := hasRequiredVersionOnMachines(appName, machines, "", backupVersion, ""); err != nil {
+	// Resolve the leader
+	leader, err := pickLeader(ctx, machines)
+	if err != nil {
 		return err
 	}
 
-	// Resolve the leader
-	leader, err := pickLeader(ctx, machines)
+	if !IsFlex(leader) {
+		return fmt.Errorf("backups are only supported on Flexclusters")
+	}
+
+	enabled, err := isBackupEnabled(ctx, appName)
 	if err != nil {
 		return err
 	}
 
+	if !enabled {
+		return fmt.Errorf("backups are not enabled. Run `fly pg backup enable -a %s` to enable them", appName)
+	}
+
+	// Ensure the the app has the required flex version.
+	if err := hasRequiredVersionOnMachines(appName, machines, "", backupVersion, ""); err != nil {
+		return handleMalformedVersionErrors(appName, err)
```

Comment:
We should just handle this within `hasRequiredVersionOnMachines`:

E.G. https://github.com/superfly/flyctl/blob/f4ae6f03c144e4d5ccbe454172c4d3bea6b1d85b/internal/command/postgres/postgres.go#L69

Reply:
Ah, good call. Fixing.

Reply:
An empty version string is an indication of incompatibility, right?

Reply:
The line I linked to isn't actually checking the image version, though. It's checking the version of the empty string being passed in as an argument. So this "Malformed version" error isn't coming from the image version at all--it comes from the fact that we're passing empty strings in as the versions to check against. For this PR it doesn't strictly matter since we're already checking the cluster type first, but it's definitely a bit misleading when it does get hit.

#3805 should make this a bit clearer. If that seems like a good path forward, I can merge it and rewrite this PR to use those version checks instead. But I think this shows how the logic here is getting harder to reason about and needs a bit of cleanup.

Reply:
We're now routing all the Flex-related version checks through `hasRequiredFlexVersionOnMachines` which, in theory, should only return "Malformed version" errors if the cluster is actually Flex. It also returns a better error message for this case, so the old helper is gone.

Agreed that we could handle errors better for Stolon and...whatever the other option we offered was, but that feels like a bigger effort, and I'd want to be sure we're comfortable committing to it before putting more time into it.